doc/LCIO2EDM4hep.md

```diff
@@ -0,0 +1,95 @@
+The order in which the data is converted does not matter because converting data and resolving relations are two differet steps that are carried out in sequence. 
+Subset collections are  handled similar to relations using the fuction [`fillSubset`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h). Internally this simply forwards to [`handleSubsetColl`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h) which handles all the type details and can obviously also be used directly.
+
+The OneToMany and OneToOne Relations can be resolved using [`resolveRelations`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h).  There is a resolveRelations function for each type. `LCRelation` only exist in LCIOand there conversion is limited to what is available in EDM4hep. They use the `"FromType"` and `"ToType"` collection parameters to get the necessary type information. 
+
+The AssociationCollections in EDM4hep are then created using [`createAssociations`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h). 
+
+`CaloHitContributions` are treated separately, since they are part of the SimCalorimeterHits in LCIO while being a their own data type in EDM4hep. They are created by [`createCaloHitContributions`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h).
+
+The EventHeader Colletion can be created using [`EventHeaderCollection`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h).
+
+Particle IDs are converted during the conversion of the the reconstructed Particle collection.
+
+
+Converting an entire event can be done calling the [`convertEvent`](../k4EDM4hep2LcioConv/include/k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h).
+
+## Example for a ReconstructedParticle Collection
+```cpp
+//the structs defined in the header file are used for the maps linking Lcio particles to there EDM counterparts.
+
+#include "k4EDM4hep2LcioConv/k4Lcio2EDM4hepConv.h"  
+
+auto convertedReconstructedParticleCollection = convertReconstructedParticle(name, LCCollection, typeMapping.recoParticles, typeMapping.particleIDs)
+
+//If the relations to other data types are supposed to be converted it is necessary that these are converted aswell by calling their convert function.
+//This needs to be done in order to fill the maps used in setting the relations.
+//For a collction of the type reconstructedparticle those are the vertex, cluster and track collections containing the data related to the reconstructed particles. 
+
+//next step is resolving the relations.
+resolveRelationsRecoParticle(
+      typeMapping.recoParticles, typeMapping.vertices, typeMapping.clusters, typeMapping.tracks);
```

Comment:
In this case it might be easier to just call the "top level" function that just takes the `typeMapping` (since that is also what is mentioned in the text above).